### PR TITLE
Kafka metrics for the channel

### PR DESCRIPTION
**Pull request by matzew**
Helps knative/eventing#1693.

## Proposed Changes

  * adding metrics for the channel

/assign @nachocano

### CODE REVIEW

File: kafka/channel/cmd/channel_dispatcher/main.go [hunk 1]
```diff
@@ -179,3 +186,8 @@ func getLoggingConfigOrDie() map[string]string {
 	}
 	return cm
 }
+
+func flush(logger *zap.SugaredLogger) {
+	_ = logger.Sync()
+	metrics.FlushExporter()
+}
```
Comment (mattmoor-sockpuppet) [inline kafka/channel/cmd/channel_dispatcher/main.go]: Add trailing newline:
```suggestion
}

```

File: kafka/channel/cmd/webhook/main.go [hunk 2]
```diff
@@ -111,3 +121,9 @@ func main() {
 	}
 	logger.Infow("Kafka webhook stopping")
 }
+
+
+func flush(logger *zap.SugaredLogger) {
+	_ = logger.Sync()
+	metrics.FlushExporter()
+}
```
Comment (mattmoor-sockpuppet) [inline kafka/channel/cmd/webhook/main.go]: Add trailing newline:
```suggestion
}

```

File: kafka/docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+cat << EOF | kubectl apply -f -
+apiVersion: messaging.knative.dev/v1alpha1
+kind: KafkaChannel
+metadata:
+  name: my-test-channel
+  namespace: default
+EOF
+```
+
+Access KafkaChannel controller metrics [http://localhost:9091/metrics](http://localhost:9091/metrics).
```
Comment (mattmoor-sockpuppet) [inline kafka/docs/metrics.md]: Add trailing newline:
```suggestion
Access KafkaChannel controller metrics [http://localhost:9091/metrics](http://localhost:9091/metrics).

```

File: kafka/channel/cmd/channel_dispatcher/main.go (hunk 1 above)
Comment (mattmoor-sockpuppet) [inline kafka/channel/cmd/channel_dispatcher/main.go]: Add trailing newline:
```suggestion
}

```

File: kafka/channel/cmd/webhook/main.go [hunk 4]
```diff
@@ -18,6 +18,7 @@ package main
 import (
 	"context"
 	"flag"
+	"knative.dev/pkg/metrics"
```
Comment (nachocano) [inline kafka/channel/cmd/webhook/main.go]: goimports?

File: kafka/channel/cmd/webhook/main.go [hunk 5]
```diff
@@ -69,6 +74,11 @@ func main() {
 
 	configMapWatcher.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, logconfig.WebhookName()))
 
+	// Watch the observability config map
+	configMapWatcher.Watch(metrics.ConfigMapName(), metrics.UpdateExporterFromConfigMap(component, logger))
+
+
```
Comment (nachocano) [inline kafka/channel/cmd/webhook/main.go]: remove a couple of these empty lines?

File: kafka/docs/metrics.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+Do a port-forward to access Prometheus targets endpoint:
+
+```bash
+kubectl port-forward -n knative-monitoring \
+   $(kubectl get pods -n knative-monitoring \
+   --selector=app=prometheus --output=jsonpath="{.items[0].metadata.name}") \
+   9090
+```
+
+Access Prometheus targets endpoint [http://localhost:9090/targets](http://localhost:9090/targets). Now, the **kafkachannel_controller**, **kafkachannel_dispatcher** and **kafkachannel_dispatcher** jobs should be up.
```
Comment (nachocano) [inline kafka/docs/metrics.md]: and **kafkachannel_webhook** ?

**Comment by nachocano** [comment]
@matzew very trivial comments and good to go! 
Thanks for adding this!!!

**Comment by matzew** [comment]
@nachocano thanks for the review!

I've patched the PR

**Pull request merged**